sci-libs/pcl/pcl-9999.ebuild

```diff
@@ -69,37 +64,77 @@ REQUIRED_USE="
 "
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-1.12.1-allow-configuration-of-install-dirs.patch
-	"${FILESDIR}"/${PN}-1.12.1-fix-hardcoded-relative-directory-of-the-installed-cmake-files.patch
+	"${FILESDIR}/${PN}-1.12.1-allow-configuration-of-install-dirs.patch"
+	"${FILESDIR}/${PN}-1.12.1-fix-hardcoded-relative-directory-of-the-installed-cmake-files.patch"
 )
 
-src_prepare() {
-	if use cuda; then
-		cuda_src_prepare
-		cuda_add_sandbox -w
-	fi
+cuda_set_CUDAHOSTCXX() {
+	local compiler
+	tc-is-gcc && compiler="gcc"
+	tc-is-clang && compiler="clang"
+	[[ -z "$compiler" ]] && die "no compiler specified"
+
+	local package="sys-devel/${compiler}"
+	local version="${package}"
+	local CUDAHOSTCXX_test
+	while
+		CUDAHOSTCXX="${CUDAHOSTCXX_test}"
+		version=$(best_version "${version}")
+		if [[ -z "${version}" ]]; then
+			if [[ -z "${CUDAHOSTCXX}" ]]; then
+				die "could not find supported version of ${package}"
+			fi
+			break
+		fi
+		CUDAHOSTCXX_test="$(
+			dirname "$(
+				realpath "$(
+					which "${compiler}-$(echo "${version}" | grep -oP "(?<=${package}-)[0-9]*")"
+				)"
+			)"
+		)"
+		version="<${version}"
+	do ! echo "int main(){}" | nvcc "-ccbin ${CUDAHOSTCXX_test}" - -x cu &>/dev/null; done
 
+	export CUDAHOSTCXX
+}
+
+cuda_get_host_arch() {
+	[[ -z "${CUDAARCHS}" ]] && einfo "trying to determine host CUDAARCHS"
+	: "${CUDAARCHS:=$(__nvcc_device_query)}"
```

Comment:
What provides this function? It looks like it is supposed to be an internal function.

Reply:
Right, but what provides `__nvcc_device_query`? The underscores make it looks like an internal function.

Reply:
`__nvcc_device_query` is part of `dev-util/nvidia-cuda-toolkit`.